staging/src/k8s.io/apiserver/pkg/endpoints/handlers/patch.go

```diff
@@ -438,15 +478,31 @@ func (p *applyPatcher) applyPatchToCurrentObject(obj runtime.Object) (runtime.Ob
 		return nil, errors.NewBadRequest(fmt.Sprintf("error decoding YAML: %v", err))
 	}
 
-	return p.fieldManager.Apply(obj, patchObj, p.options.FieldManager, force)
+	if p.validationDirective == metav1.FieldValidationStrict {
+		strictPatchObj := &unstructured.Unstructured{Object: map[string]interface{}{}}
+		if err := yaml.UnmarshalStrict(p.patch, &strictPatchObj.Object); err != nil {
+			return nil, errors.NewBadRequest(fmt.Sprintf("error strict decoding YAML: %v", err))
```

Comment:
if Object is unstructured, we'll catch duplicate fields here, but not unknown fields... a unit test sending an apply patch containing duplicate fields and unknown fields to a custom resource type should ensure we get back *both* the duplicate and unknown fields errors ... otherwise, a kubectl user risks getting back a failure that doesn't capture everything that was wrong with their request

Reply:
sounds good